src/main/java/me/zeroeightsix/kami/event/ForgeEventProcessor.kt

```diff
@@ -33,48 +33,63 @@ object ForgeEventProcessor {
 
     @SubscribeEvent
     fun onTick(event: TickEvent.ClientTickEvent) {
-        KamiMod.EVENT_BUS.post(event)
+        KamiEventBus.post(event)
+
+        if (mc.world != null && mc.player != null) {
+            SafeTickEvent(event.phase).also {
+                KamiEventBus.post(it)
+                ModuleManager.onUpdate(it)
+            }
+        }
 
         if (event.phase == TickEvent.Phase.END) {
             if (prevWidth != mc.displayWidth || prevHeight != mc.displayHeight) {
                 prevWidth = mc.displayWidth
                 prevHeight = mc.displayHeight
-                val resolutionUpdateEvent = ResolutionUpdateEvent(mc.displayWidth, mc.displayHeight)
-                KamiMod.EVENT_BUS.post(resolutionUpdateEvent)
+                KamiEventBus.post(ResolutionUpdateEvent(mc.displayWidth, mc.displayHeight))
                 for (component in KamiMod.getInstance().guiManager.children) {
                     if (component !is Frame) continue
                     KamiGUI.dock(component)
                 }
             }
-        }
-
-        if (mc.world != null && mc.player != null) {
-            ModuleManager.onUpdate()
```

Comment:
that moment when onUpdate() was being called twice :moyai:

Reply:
That is deletion 🐒